internal/testrunner/runners/pipeline/runner.go

```diff
@@ -248,6 +250,19 @@ func (r *runner) verifyResults(testCaseFile string, config *testConfig, result *
 	return nil
 }
 
+// stripEmptyTestResults function removes events which are empty strings. These strings can represent
+// documents processed by a pipeline which potentially used a "drop" processor (to drop the event at all).
+func stripEmptyTestResults(result *testResult) *testResult {
+	var tr testResult
+	for _, event := range result.events {
+		if len(event) == 0 {
```

Comment:
I think we are expecting `event` to be `nil` for empty test results. If so, should we make this check stricter by explicitly checking for `nil`? Otherwise theoretically `[]` would also match (although I know practically we would not see such an event).

```suggestion
		if event == nil {
```

Reply:
Fixed